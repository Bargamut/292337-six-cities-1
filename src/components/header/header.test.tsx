import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Header from './header';

const mock = {
  user: {
    id: 1,
    email: `test@email.ru`,
    name: `Name Surname`,
    avatarUrl: ``,
    isPro: false
  },
  onClickSignIn: jest.fn()
};

describe(`SignIn renders`, () => {
  it(`SignIn correctly renders without authorization required`, () => {
    const {
      onClickSignIn
    } = mock;

    const header = renderer.create(<Header
      isAuthorizationRequired={true}
      user={{}}
      onClickSignIn={onClickSignIn}
    />).toJSON();

    expect(header).toMatchSnapshot();
  });

  it(`SignIn correctly renders with authorization required`, () => {
    const {
      user,
      onClickSignIn
    } = mock;

    const header = renderer.create(<Header
      isAuthorizationRequired={false}
      user={user}
      onClickSignIn={onClickSignIn}
    />).toJSON();

    expect(header).toMatchSnapshot();
  });
});
