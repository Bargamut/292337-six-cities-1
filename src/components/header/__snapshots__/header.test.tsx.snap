// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders SignIn correctly renders with authorization required 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="#"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={
                  Object {
                    "backgroundImage": "url(https://es31-server.appspot.com/six-cities)",
                  }
                }
              />
              <span
                className="header__user-name user__name"
              >
                test@email.ru
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`SignIn renders SignIn correctly renders without authorization required 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="#"
              onClick={[MockFunction]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                Sign In
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
